android/src/main/java/com/amplitude/android/utilities/AndroidNetworkConnectivityChecker.kt

```diff
@@ -35,23 +35,28 @@ class AndroidNetworkConnectivityChecker(private val context: Context, private va
             return true
         }
 
-        val cm = context.getSystemService(Context.CONNECTIVITY_SERVICE)
-        if (cm is ConnectivityManager) {
-            if (isMarshmallowAndAbove) {
-                val network = cm.activeNetwork ?: return false
-                val capabilities = cm.getNetworkCapabilities(network) ?: return false
-
-                return capabilities.hasTransport(NetworkCapabilities.TRANSPORT_WIFI) ||
-                    capabilities.hasTransport(NetworkCapabilities.TRANSPORT_CELLULAR)
+        val result = ExceptionUtils.safeInvoke(logger) {
```

Comment:
Could you please leave a comment pointing to the Android issue that this fixes? (Here and anywhere else that we're working around the same bug).

Reply:
@crleona why do we want a comment in code if source control already maintains the history?

Reply:
We didn't catch this the first time, so it's non-obvious. Why should I round trip through version control versus having that information right in front of me in the source code?

Reply:
Comments are cheap and no harm adding the links directly to save the extra round trip. Added more comments.